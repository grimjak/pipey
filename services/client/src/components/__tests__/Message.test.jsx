import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer'

import Message from '../Message';

const successRemoveMessage = jest.fn();
const dangerRemoveMessage = jest.fn();


const testDate = [
    {
        messageName: 'Hello, world!',
        messageType: 'success',
        removeMessage: successRemoveMessage,
    },
    {
        messageName: 'Hello, world!',
        messageType: 'danger',
        removeMessage: dangerRemoveMessage, 
    }
]

describe('When given a message', () => {
    testDate.forEach((e1) => {
        const component = <Message {...e1} />;
        it(`${e1.messageType} Message renders properly`, () => {
            const wrapper = shallow(component);
            const element = wrapper.find(`.notification.is-${e1.messageType}`);
            expect(element.length).toBe(1);
            const span = wrapper.find('span');
            expect(span.length).toBe(1);
            expect(span.get(0).props.children).toContain(
                e1.messageName);
            const button = wrapper.find('button');
            expect(button.length).toBe(1);
            expect(e1.removeMessage).toHaveBeenCalledTimes(0);
            button.simulate('click');
            expect(e1.removeMessage).toHaveBeenCalledTimes(1);
        });

        it(`${e1.messageType} Message renders a snapshot properly`, () => {
            const tree = renderer.create(
                <Message {...e1} />
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
});