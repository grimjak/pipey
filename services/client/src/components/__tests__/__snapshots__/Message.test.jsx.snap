// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When given a message danger Message renders a snapshot properly 1`] = `
<div
  className="notification is-danger"
>
  <button
    className="delete"
    onClick={[Function]}
  />
  <span>
    Hello, world!
  </span>
</div>
`;

exports[`When given a message success Message renders a snapshot properly 1`] = `
<div
  className="notification is-success"
>
  <button
    className="delete"
    onClick={[Function]}
  />
  <span>
    Hello, world!
  </span>
</div>
`;
